### DISCUSSION TITLE
Trying to create text backgrounds like tiktok and instagram

### DISCUSSION THREAD

**Discussion by ericvaroto** [dlemstra/Magick.NET#1705] [Help] [answered]
Hi, I'm trying to create text backgrounds like those found on tiktoks

Original Image:
![barn](https://github.com/user-attachments/assets/7aae71f6-2124-475f-8dd2-526dd01156a9)

I've found this Magick CLI command, but I can't convert it to C# code.
Can someone help me?

`magick -size 200x -font DejaVu-Sans -background none -undercolor white -gravity Center -pointsize 30 -fill black -stroke none caption:"Genuinely so nice and simple" -bordercolor none -border 12 ( +clone -morphology dilate disk:5 ) +swap -compose over -composite barn.jpg +swap -composite result7.png`

Image with text after command:
![result7](https://github.com/user-attachments/assets/359412e3-5df6-448d-98fa-16f23349d220)

**Comment by dlemstra** [comment]
Where are you stuck and what have you tried so far?

**Reply by ericvaroto** [reply]
Hi,
I'm trying to generate a caption with a white background only behind the words, like the example I've sent before. After that, I will add this generates caption (composite) at another image.

**Comment by ericvaroto** [comment, marked as answer]
After many attempts, I managed to generate the image identical to the command. Here's the C# code:

```
public static async Task GenerateCaptionOverImageAsync()
{
    // Generate the caption
    using var caption = new MagickImage("caption:Genuinely so nice and simple", new MagickReadSettings
    {
        Width = 200,
        Font = "DejaVu-Sans",
        BackgroundColor = MagickColors.None,
        TextUnderColor = MagickColors.White,
        TextGravity = Gravity.Center,
        FontPointsize = 30,
        FillColor = MagickColors.Black,
        StrokeColor = MagickColors.None,
    });

    caption.BorderColor = MagickColors.None;
    caption.Border(12);

    // Clone caption to create back color
    var captionBackground = caption.Clone();
    captionBackground.Morphology(MorphologyMethod.Dilate, Kernel.Disk, "5");

    // Merge the caption and the back color, putting caption over back color
    captionBackground.Composite(caption, Gravity.Center, CompositeOperator.Over);

    // Get the background image
    using var imageBackground = new MagickImage(@"path\background-image.jpg");

    // Merge background image with caption and back color
    imageBackground.Composite(captionBackground, Gravity.Center, CompositeOperator.Over);

    // Save result
    await imageBackground.WriteAsync(@"path\result-dotnet.png");
}
```
